oct_converter/readers/img.py

```diff
@@ -46,7 +48,62 @@ def read_oct_volume(
                 interlaced = np.rot90(interlaced, axes=(0, 1))
                 volume = interlaced
 
+        meta = self.get_metadata_from_filename()
+        lat_map = {"OD": "R", "OS": "L", None: ""}
+
         oct_volume = OCTVolumeWithMetaData(
-            [volume[:, :, i] for i in range(volume.shape[2])]
+            [volume[:, :, i] for i in range(volume.shape[2])],
+            patient_id=meta.get("patient_id"),
+            acquisition_date=meta.get("acq"),
+            laterality=lat_map[meta.get("lat", None)],
+            metadata=meta,
         )
         return oct_volume
+
+    def get_metadata_from_filename(self) -> dict:
+        """Attempts to find metadata within the filename
+
+        Returns:
+            meta: dict of information extracted from filename
+        """
+        filename = Path(self.filepath).name
+        meta = {}
+        meta["patient_id"] = (
+            re.search(r"^P\d+", filename).group(0)
+            if re.search(r"^P\d+", filename)
+            else None
+        )
+        acq = (
+            list(
+                re.search(
+                    r"(?P<m>\d{1,2})-(?P<d>\d{1,2})-(?P<y>\d{4})_(?P<h>\d{1,2})-(?P<M>\d{1,2})-(?P<s>\d{1,2})",
+                    filename,
+                ).groups()
+            )
+            if re.search(
+                r"(?P<m>\d{1,2})-(?P<d>\d{1,2})-(?P<y>\d{4})_(?P<h>\d{1,2})-(?P<M>\d{1,2})-(?P<s>\d{1,2})",
+                filename,
+            )
+            else None
+        )
+        if acq:
+            meta["acq"] = datetime(
```

Comment:
super minor but could we use acquisition_date, laterality etc as the keys here just to be more consistent with the other readers and increase readability?